### PR TITLE
Rewrite

### PR DESCRIPTION
**Pull request by 1000ch**
- [x] rewrite with es6

related to https://github.com/AtomLinter/linter-plus/issues/58

### CODE REVIEW

**Comment by steelbrain** [comment]
@iam4x wanna take this one too?

**Comment by steelbrain** [comment]
@1000ch just a side note, but an important one, Atom, like any other chrome-based app has a single thread for rendering, and doing cpu-intensive operations on that process can significantly lag the editor for the user. I would recommend you jsonhint logic to another file called `cli.js` and spawn that file from this process using `atom-linter`'s `execNode`.

**Comment by 1000ch** [comment]
@steelbrain Ah, I understand. But it is little unwillingly :O

**Comment by iam4x** [comment]
@steelbrain Your statement is true, but it depends a lot. For instance we had to rollback `linter-eslint` to use an API because it feels much faster than spawning a process child. I think the process is almost the same for `linter-json`.

I will take this also.

File: index.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+        try {
+          jsonlint.parse(text);
+        } catch (e) {
+
+          let line = Number(e.message.match(this.regex)[1]);
+          let column = 0;
+
+          return [{
+            type: 'error',
```
Comment (steelbrain) [inline index.js]: How about you uppercase this to `Error`

File: index.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            text: e.message,
+            filePath: path,
+            range: new Range([line, column], [line, column])
```
Comment (steelbrain) [inline index.js]: You might wanna make this `range: new Range([line, column], [line, column + 1])`
as a decoration with `0` char diff will never show a bubble for the user.

**Comment by steelbrain** [comment]
@iam4x Great, I'll leave the rest to you!

**Comment by k2b6s9j** [comment]
The code looks good to me, on initial review. I can merge and publish if you say it's ready @1000ch.

This would fix #17

**Comment by 1000ch** [comment]
Please!

**Comment by k2b6s9j** [comment]
This will be the next major release.

**Pull request merged**